app/src/main/java/ch/epfl/sdp/ui/event/CreateEventFragment.java

```diff
@@ -0,0 +1,78 @@
+package ch.epfl.sdp.ui.event;
+
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.Fragment;
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.ViewModelProvider;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Toast;
+
+import java.text.ParseException;
+
+import ch.epfl.sdp.R;
+import ch.epfl.sdp.databinding.CreateEventFragmentBinding;
+
+
+public class CreateEventFragment extends Fragment {
+    private EventViewModel viewModel;
+    private CreateEventFragmentBinding binding;
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        viewModel = new ViewModelProvider(this).get(EventViewModel.class);
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+                             @Nullable Bundle savedInstanceState) {
+        binding = CreateEventFragmentBinding.inflate(inflater, container, false);
+        View view = binding.getRoot();
+
+        binding.createButton.setOnClickListener(v -> {
```

Comment:
Si tu veux réduire la taille de la fonction tu peux dire que ta classe implémente OnClickListener, définir la méthode qu'il faut et gérer tous les boutons dans cette méthode en faisant un switch sur les id des boutons. Ensuite tu peux juste faire setOnClickListener(this).

Reply:
Du coup on perd le view binding ici?

Reply:
ça te fait utiliser les id des view mais c'est pas très grave puisque tu les utilises pas, tu testes une égalité